product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.1-Modify-payload-by-adding-a-child-using-enrich-mediator/src/test/java/org.wso2.carbon.ei.scenario.test/AddChildToXMLTest.java

```diff
@@ -89,6 +89,50 @@ public void testPassThroughProxyTemplate() throws IOException, XMLStreamExceptio
         assertResponse(response, expectedResponse);
     }
 
+    /**
+     * This test is to verify if payload can be modified by adding payload stored in a property (OM type)
+     * as a child to the message body
+     */
+    @Test(description = "1.6.1.4")
+    public void addPayloadStoredInPropertyAsChild() throws IOException, XMLStreamException {
+        String url = getProxyServiceURLHttp("1_6_1_4_Proxy_addPayloadStoredInPropertyAsChild");
+        String testCaseId = "1.6.1.4";
+        String request =
+                "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+                        + "<soap:Envelope xmlns:soap=\"http://schemas.xmlsoap.org/soap/envelope/\">\n"
+                        + "   <soap:Header />\n"
+                        + "   <soap:Body>\n"
+                        + "      <Company>\n"
+                        + "         <Name>WSO2</Name>\n"
+                        + "         <Location>Colombo, Sri Lanka</Location>\n"
+                        + "         <Dept>Engineering</Dept>\n"
+                        + "      </Company>\n"
+                        + "   </soap:Body>\n"
+                        + "</soap:Envelope>";
+
+        String expectedResponse =
+                "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+                        + "<soap:Envelope xmlns:soap=\"http://schemas.xmlsoap.org/soap/envelope/\">\n"
+                        + "   <soap:Body>\n"
+                        + "      <Company>\n"
+                        + "         <Name>WSO2</Name>\n"
+                        + "         <Location>Colombo, Sri Lanka</Location>\n"
+                        + "         <Dept>\n"
+                        + "            Engineering\n"
+                        + "            <Employee>\n"
```

Comment:
We should embed the property content as a child, not append to child content

Reply:
This is done.

Reply:
This is done